nautobot/dcim/models/locations.py

```diff
@@ -88,14 +88,15 @@ def clean(self):
     @property
     def display(self):
         # prefix with parent name allows location type name identity if there are the same name exists under different parents.
-        display_str = self.name
-        parent = self.parent
-        for _ in range(10):  # "10 levels iteration"
-            if parent is None:
-                break
-            display_str = f"{parent.name} → {display_str}"
-            parent = parent.parent
-        return display_str
+        display_str = ""
+        try:
+            for ancestor in self.ancestors():
+                display_str += ancestor.name
+                display_str += " → "
+            display_str += self.name
+            return display_str
+        except ObjectDoesNotExist:
+            return display_str
```

Comment:
This reads as backwards to me - if we hit an ObjectDoesNotExist on one of the ancestors, we may return a partially-constructed `display_str` that includes some ancestors but not `self.name` (since that's the last thing we add)?